pkg/reconciler/pubsub/pullsubscription/resources/receive_adapter.go

```diff
@@ -163,6 +163,17 @@ func makeReceiveAdapterPodSpec(ctx context.Context, args *ReceiveAdapterArgs) *c
 		}
 	}
 
+	// If k8s service account is specified, use that service account as credential.
```

Comment:
Same change in `pkg/reconciler/intevents/pullsubscription/resources/receive_adapter.go`.